core/src/main/scala/org/apache/spark/deploy/history/ApplicationHistoryProvider.scala

```diff
@@ -73,4 +102,18 @@ private[history] abstract class ApplicationHistoryProvider {
   @throws(classOf[SparkException])
   def writeEventLogs(appId: String, attemptId: Option[String], zipStream: ZipOutputStream): Unit
 
+  /**
+   * Probe for an update to an (incompleted) application
+   * @param appId application ID
+   * @param attemptId optional attempt ID
+   * @param updateState state information needed by the provider to determine age
+   * @return true if the application has been updated
+   */
+  def isUpdated(
+      appId: String,
+      attemptId: Option[String],
+      updateState: Option[HistoryProviderUpdateState]): Boolean = {
+    false
```

Comment:
why provide a default implementation here?  this is unlikely to be the right implementation, just means somebody implementing this interface is more likely to overlook supplying the right definition here.